crates/core/src/lib.rs

```diff
@@ -399,6 +399,14 @@ impl<T: OutboundWasiHttpHandler + Send + Sync> Engine<T> {
         Ok(InstancePre { inner })
     }
 
+    /// Returns the type of this component to reflect on it at runtime.
+    #[instrument(skip_all)]
```

Comment:
Are you intentionally wanting this to be instrumented? Only reason I can think is if `substituted_component_type` has a chance of being very slow?

Reply:
Oh I don't know what this attribute does, I was just copying `instantiate_pre` above. I'll got ahead and remove it if it's related to performance measuring.

Reply:
Ah yeah its a `tracing` thing that wraps calls to the function in a new span. We're going to start caring more about these soon because of https://github.com/fermyon/spin/pull/2348